Simplified/SignInLogic/NYPLSignInBusinessLogic+Adept.swift

```diff
@@ -76,51 +77,33 @@ extension NYPLSignInBusinessLogic {
       VendorID: \(vendor ?? "N/A")
       """)
 
-    drmAuthorizerAdobe?
-      .authorize(withVendorID: vendor,
-                 username: username,
-                 password: password) { success, error, deviceID, userID in
-
-                  // make sure to cancel the previously scheduled selector
```

Comment:
if I understand correctly, I think you moved all this code to the shared NYPLSignInBusinessLogic+DRM.swift file, correct? ideally this code should stay here, because it's related to Adobe only. The only thing that doesn't really belong here is the call to `finalizeSignIn`, because that's shared with Axis, and we only want to call that once. So here's what I'd do:
-  I'd revert back this `drmAuthorizeAdobe` function and instead of calling `finalizeSignIn` I'd simply call a `completion(success, error)`. This `completion` variable is of course an input param to drmAuthorizeAdobe. That's it here.
- Then proceeding bottom up, the caller `authorizeWithAdobe` will have some compile errors to fix....
- and finally the call to `authorizeWithAdobe` inside `drmAuthorizeUserData` will specify the closure literal with the `finalizeSignIn` call:
```
authorizeWithAdobe(userProfile: profileDoc,
                   loggingContext: loggingContext) { success, err in
  finalizeSignIn(forDRMAuthorization: success, error: err)
}
```
- Same for Axis, but as we know the completion will have a preprocessor condition:
```
drmAuthorizeAxis(username: profileDoc.authorizationIdentifier ?? "",
                 password: profileDoc.authorizationIdentifier,
                 loggingContext: loggingContext) { success, err in
#if !FEATURE_DRM_CONNECTOR
  finalizeSignIn(forDRMAuthorization: success, error: err)
#endif
    }
```
While this is still not great (`#if` stuff never is) at least this way we'll have one function (`drmAuthorizeUserData`) explicitly in charge of the `finalizeSignIn` logic, instead of having the lower function be aware of what's above them. The 2 conditional calls to `finalizeSignIn` will also be close to each other.

If we instead had 2 separate functions as you added to the bottom of NYPLSignInBusinessLogic+DRM.swift, there will still be interdependencies between the 2 functions and we'll have to jump from one to the other to understand exactly what's going on.

Let me know what you think or if I am missing something.